### PR TITLE
[backend] improve waiting time for obas scenario generation with AI (#8585)

### LINKED ISSUE #8585: time out on Simulate report with IA and Email for OpenBas
## Description

- Time out after 1 min when you simulate with IA and Email for OpenBas

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Create an Report
2. Go to this report
3. Click on Simulate (open a drawer)
4. Select `Use AI`
5. Select `Simulated (emails)`
6. Click on Generate

## Expected Output

Works

## Actual Output

Time out after 1 min!

## Additional information

- With this configuration, we generate multiple email with AI prompt. At each loop, we have 4 prompt (1 for email content, 1 for email subject, and the same for a second email). The 2 prompts for email subjects are fast, but the 2 prompts for email contents are very slow. The multiple generation are slow, more than 1 minute, and it's over the maximum delay authorized for a query.
- Current delay : approximatly 1m30s
- If you update prompts to generate only 1 paragraph (in place of 3) of 3 lines, we can reduce at : approximatly 1m (not enough)
- On local it works, but the limit is on deploy platforms.
- Issue OpenBas : https://github.com/OpenBAS-Platform/openbas/issues/1321
- Branch OpenCTI with console.log to test : https://github.com/OpenCTI-Platform/opencti/compare/master...issue/simulate-for-obas
- Maybe improve prompt ?
- Maybe have a timeout after 2 min ?

### PR DESCRIPTION
**Pull request by aHenryJard**
### Proposed changes

* Use promises to send all scenario generation at once, instead of one after the other.
* Add a warn log in case the time is over 2 minutes.

Refactor:
- some renaming to know which part is an OBAS call, and which part is an AI call.

### Related issues
* closes #8585 
* 
### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-graphql/src/database/ai-llm.ts [hunk 1]
```diff
@@ -31,11 +31,12 @@ if (AI_ENABLED && AI_TOKEN) {
 }
 
 export const queryMistralAi = async (busId: string | null, question: string, user: AuthUser) => {
+  // const startingTime = new Date().getTime();
```
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/src/database/ai-llm.ts]:
```suggestion
```

File: opencti-platform/opencti-graphql/src/modules/xtm/xtm-domain.js [hunk 2]
```diff
@@ -195,168 +205,210 @@ export const generateOpenBasScenario = async (context, user, stixCoreObject, att
...
+  const totalTime = endingTime - startingTime;
+  if (totalTime > 120000) {
+    logApp.warn(`[OPENCTI-MODULE][XTM] Long scenario generation time. Generating ${createAndInjectScenarioPromises.length} emails took ${totalTime} ms`, { useAI, simulationType });
   }
+  logApp.info(`[OPENCTI-MODULE][XTM] Generating ${createAndInjectScenarioPromises.length} emails took ${totalTime} ms`, { useAI, simulationType });
   return `${XTM_OPENBAS_URL}/admin/scenarios/${obasScenario.scenario_id}/injects`;
 };
 
 export const generateContainerScenario = async (context, user, args) => {
+  // ici
```
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/src/modules/xtm/xtm-domain.js]:
```suggestion
```

File: opencti-platform/opencti-graphql/src/database/ai-llm.ts (hunk 1 above)
Comment (frapuks) [reply opencti-platform/opencti-graphql/src/database/ai-llm.ts]: Would you like to keep this comment ?

File: opencti-platform/opencti-graphql/src/modules/xtm/xtm-domain.js (hunk 2 above)
Comment (frapuks) [reply opencti-platform/opencti-graphql/src/modules/xtm/xtm-domain.js]: Would you like to keep this comment ?

**Review by frapuks — APPROVED** [review]
Ok for me, just clarify the comments

File: opencti-platform/opencti-graphql/src/database/ai-llm.ts (hunk 1 above)
Comment (aHenryJard) [reply opencti-platform/opencti-graphql/src/database/ai-llm.ts]: no I forgot :)

**Pull request merged**